pkg/executor/adapter.go

```diff
@@ -241,6 +242,21 @@ func (a *recordSet) OnFetchReturned() {
 	a.stmt.LogSlowQuery(a.txnStartTS, len(a.lastErrs) == 0, true)
 }
 
+// Detach creates a new `RecordSet` which doesn't depend on the current session context.
+func (a *recordSet) Detach() (sqlexec.RecordSet, error) {
+	cteStorage := a.stmt.Ctx.GetSessionVars().StmtCtx.CTEStorageMap
```

Comment:
Can we return an error when `cteStorage` is not empty because CTE is not currently supported.

In the future, if we want to support CTE,  I think we should copy this map

Reply:
Done.